src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GCNVPostProcessor.java

```diff
@@ -0,0 +1,197 @@
+package org.broadinstitute.hellbender.tools.copynumber.gcnv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.vcf.*;
+import org.apache.commons.math3.util.FastMath;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class for single sample gCNV postprocessing
+ */
+public class GCNVPostProcessor {
+
+    /**
+     * VCF miscellaneous header keys
+     */
+    public static final String SAMPLE_NAME = "sample_name";
+
+    /**
+     * VCF FORMAT header keys
+     */
+    public static final String CN = "CN";
+    public static final String CNLP = "CNLP";
+    public static final String CNQ = "CNQ";
+
+    private final List<Allele> alleles;
+    private final IntegerCopyNumberStateCollection integerCopyNumberStateCollection;
+    private final String sampleName;
+    private final VariantContextWriter outputWriter;
+    private final SAMSequenceDictionary samSequenceDictionary;
+
+    /**
+     * We store the previously analyzed record for validating that records are ordered according to
+     * the sequence dictionary
+     */
+    private CopyNumberPosteriorLocatableRecord lastPosteriorRecord = null;
+
+    /**
+     * gCNV postprocessor constructor
+     *
+     * @param outputWriter variant context writer
+     * @param integerCopyNumberStateCollection collection of copy number states considered by post processor
+     * @param sampleName sample name
+     * @param samSequenceDictionary sequence dictionary used for interval order validation
+     */
+    GCNVPostProcessor(final VariantContextWriter outputWriter,
+                                final IntegerCopyNumberStateCollection integerCopyNumberStateCollection,
+                                final String sampleName,
+                                final SAMSequenceDictionary samSequenceDictionary) {
+        //TODO pass a list of intervals and add progress meter
+        //ProgressMeter progressMeter = new ProgressMeter(1.0);
+        //progressMeter.start();
+        this.outputWriter = Utils.nonNull(outputWriter);
+        this.integerCopyNumberStateCollection = Utils.nonNull(integerCopyNumberStateCollection);
+        Utils.validate(integerCopyNumberStateCollection.size() > 2, "There must be at least 3 copy number states present");
```

Comment:
period at end.  (also, this is super nitpicky, but you use `>= 3` when you check this condition elsewhere!)

Reply:
Done